integration_tests/e2e/bookings.cy.ts

```diff
@@ -0,0 +1,139 @@
+import paths from '../../server/constants/paths'
+import TestData from '../../server/routes/testutils/testData'
+import BookingsPage from '../pages/bookings'
+import CancelledVisitsPage from '../pages/bookings/cancelledVisitsPage'
+import PastVisitsPage from '../pages/bookings/pastVisitsPage'
+import VisitDetailsPage from '../pages/bookings/visit'
+import HomePage from '../pages/home'
+import Page from '../pages/page'
+
+context('Bookings home page', () => {
+  const orchestrationVisitDto = TestData.orchestrationVisitDto({
+    startTimestamp: '2026-05-21T10:00:00',
+    endTimestamp: '2026-05-21T11:30:00',
+    prisonId: 'DHI',
+  })
+  const pastVisitDto = TestData.orchestrationVisitDto({
+    startTimestamp: '2023-05-30T10:00:00',
+    endTimestamp: '2023-05-30T11:30:00',
+    prisonId: 'DHI',
+  })
+  const cancelledVisitDto = TestData.orchestrationVisitDto({
+    startTimestamp: '2026-05-21T10:00:00',
+    endTimestamp: '2026-05-21T11:30:00',
+    prisonId: 'DHI',
+    outcomeStatus: 'ESTABLISHMENT_CANCELLED',
+    visitStatus: 'CANCELLED',
+  })
+
+  const prisoner = TestData.bookerPrisonerInfoDto({ prisonId: 'DHI' })
+  const bookerReference = TestData.bookerReference()
+
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubHmppsAuthToken')
+  })
+
+  it('should show Bookings home page with future visits and navigate to view the visit details', () => {
+    cy.task('stubGetBookerReference')
+    cy.task('stubGetPrisoners', { prisoners: [prisoner] })
+    cy.signIn()
```

Comment:
These three lines are now repeated in each test so could add them to the `beforeEach()` block - works fine as is though.